src/fides/api/util/consent_util.py

```diff
@@ -228,3 +247,281 @@ def get_or_create_fides_user_device_id_provided_identity(
+            )
+            validate_privacy_notice_dry_update(
+                dry_update
+            )  # Checks consent mechanism + delivery location
+            validation_updates.append(
+                existing_notice.dry_update(data=update_data.dict(exclude_unset=True))
+            )
+            # and don't include it anymore in the existing notices used for validation
+            existing_notices.pop(existing_notice.id, None)
+        else:
+            # Add the pending created record to the the temporary updates for validation
+            validation_updates.append(model(**update_data.dict(exclude_unset=True)))
+
+    # run the validation here on our proposed "dry-run" updates
+    try:
+        check_conflicting_data_uses(
+            validation_updates,
+            existing_notices.values(),
+            ignore_disabled=ignore_disabled,
+        )
+    except ValidationError as e:
+        raise HTTPException(HTTP_422_UNPROCESSABLE_ENTITY, detail=e.message)
+
+    # return the tuples of update data associated with their existing db records
+    return updates_and_existing
+
+
+def upsert_privacy_notice_templates_util(
+    db: Session,
+    template_schemas: List[Union[PrivacyNoticeWithId]],
+) -> List[PrivacyNoticeTemplate]:
+    """
+    Create or update *Privacy Notice Templates*. These are the resources that
+    Fides ships with out of the box.
+    """
+    ensure_unique_ids(template_schemas)
+    validate_notice_data_uses(template_schemas, db)  # type: ignore[arg-type]
+
+    upserts_and_existing: List[
+        Tuple[PrivacyNoticeWithId, Optional[PrivacyNoticeTemplate]]
+    ] = prepare_privacy_notice_patches(  # type: ignore[assignment]
+        template_schemas,
+        db,
+        model=PrivacyNoticeTemplate,
+    )
+
+    upserted_templates = []
+    for (
+        create_or_update_data,
+        existing_template,
+    ) in upserts_and_existing:
+        if existing_template:
+            upserted_templates.append(
+                existing_template.update(
+                    db, data=create_or_update_data.dict(exclude_unset=True)
+                )
+            )
+        else:
+            upserted_templates.append(
+                PrivacyNoticeTemplate.create(
+                    db=db,
+                    data=create_or_update_data.dict(exclude_unset=True),
+                    check_name=False,
+                )
+            )
+    return upserted_templates  # type: ignore[return-value]
+
+
+def load_default_notices_on_startup(
+    db: Session, notice_yaml_file_path: str
+) -> Tuple[List[PrivacyNoticeTemplate], List[PrivacyNotice]]:
+    """
+    On startup, loads any new PrivacyNoticeTemplates into the database, and updates
+    existing templates where applicable.
+
+    Creates Privacy Notices from these templates, if no notices are linked to the given template
+    already.  Otherwise, we skip making updates to the Notices themselves automatically.
+    """
```

Comment:
This is my opinion of how this function should behave.  I upsert the templates from the yaml file we ship with. And then, in the other table, if the templates were new, I create privacy notices from them, exposing them to the end user.  But if they aren't new, I don't update existing notices.  Customers may have updated the auto-generated notices since their creation.

Reply:
that makes sense, i think it's also how i'd expect this to behave (though i'm a lot less up to speed on the use case than you are!)

have you clarified with product that this is the desired behavior? just because this feels like business logic rather than technical constraints driving this (not that it's a bad thing) - so i'd expect us to be doing whatever product thinks is best.

Reply:
good callout - will update when I hear back from Michael

Reply:
OK I am preemptively merging this work ahead of the follow-up changes I need to make tonight in a separate ticket, but will reticket if the default behavior needs to change -

Reply:
OK confirmed with Michael that this logic is ok for now -